src/collapse/dtrace.rs

```diff
@@ -273,12 +264,18 @@ impl Folder {
                 }
             }
 
+            // Send shutdown signal.
             for _ in &handles {
```

Comment:
Actually, thinking some more about this, we don't need the `Option` at all. Just `drop(handles)` here, and the receiver will get `None` back from `rx_input.recv().ok()` (notice the lack of an `unwrap()`).

Reply:
I'll play around with this idea. Not sure what to think of it until I try it out.

Reply:
I think this still applies. By dropping `handles`, every thread would be notified that `rx_input` has been closed.